h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1488,10 +1488,12 @@ public AutoBuffer putStr( String s ) {
   }
 
   public static Object javaSerializeReadPojo(byte [] bytes) {
+    String className = nameOfClass(bytes);
```

Comment:
This is necessary for standard (non-exceptional) execution. This adds some time to the method invocation, I would rather leave it in the catch-block.